python/lsst/eochar/pixel_covar.py

```diff
@@ -0,0 +1,106 @@
+"""
+Pixel covariance calculator
+
+Compute the pixel covariance matrix on a pair of
+flat fields, taking masks into account.
+If (k,l) is the (row,col) position of a pixel,
+its pixel covariance is
+Mean(Diff(j,i)*Diff(j-l,i-k)) - Mean(Diff(j,i))*Mean(Diff(j-l,i-k))
+where Diff stands for the (flat1-flat2) array, and "*" in the
+first term is an element-wise product.
+"""
+
+from lsst.eotest.sensor.MaskedCCD import MaskedCCD
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import numpy as np
+from math import sqrt
+import os
+
+def create_diff(ccd_1, ccd_2, amp=5):
+    """
+    returns the difference of two flat images from the sensor
+    at position amp, after standard unbiasing and trimming
+    """
+    image1 = ccd_1.unbiased_and_trimmed_image(amp)
+    image2 = ccd_2.unbiased_and_trimmed_image(amp)
+
+    #subtract the two images
+    diff_image = afwImage.MaskedImageF(image1, True)
+    diff_image -= image2
+    return diff_image
+
+def variance_threshold_masking(arr, sigma):
+    """
+    simple masking based on the condition variance>sigma
+    """
+    diffmed = np.median(arr)
```

Comment:
Instead of lines 33-40, it would be more idiomatic to do instead
```
def variance_threshold_masking(fdiff, sigma):
    """
    simple masking based on the condition variance>sigma
    """
    arr = fdiff.getImage().getArray()
    stats = afwMath.makeStatistics(fdiff, afwMath.MEDIAN | afwMath.VARIANCECLIP)
    diffmed = stats.getValue(afwMath.MEDIAN)
    diffvar = stats.getValue(afwMath.VARIANCECLIP)
```
where fdiff is the differenced MaskedImage object from create_diff().